storage/remote/read_handler.go

```diff
@@ -58,7 +58,7 @@ func NewReadHandler(logger log.Logger, r prometheus.Registerer, queryable storag
 
 		queries: prometheus.NewGauge(prometheus.GaugeOpts{
 			Namespace: "prometheus",
-			Subsystem: "api", // TODO: changes to storage in Prometheus 3.0.
+			Subsystem: "storage",
```

Comment:
The read client also exposed a `remote_read_queries` metric under the `remote_storage` subsystem https://github.com/prometheus/prometheus/blob/ff0a1e5e1165caa472c522f6dc25281d7e6a483f/storage/remote/client.go#L83-L90, is this ok?

Reply:
The change now yields `prometheus_remote_read_handler_remote_read_queries`, I think we can remove the second `remote`, to make it `prometheus_remote_read_handler_read_queries` or even `prometheus_remote_read_handler_queries`. (or even get back `prometheus_remote_read_storage_queries` if we make the changes below)

I think we can leverage `3.0` to also clarify the others in `client.go`

Currently, `remoteReadQueriesTotal` yields `prometheus_remote_storage_read_queries_total`
and `remoteReadQueries` translates to `prometheus_remote_storage_remote_read_queries`.

I think `storage` has nothing to do there, and it'd be clearer if we mention `client`:

- `prometheus_remote_storage_read_queries_total` -> `prometheus_remote_read_client_queries_total`
- `prometheus_remote_storage_remote_read_queries` -> `prometheus_remote_read_client_queries`

WDYT?

Reply:
I like the client approach, LGTM for that!